src/dbHelper.ts

```diff
@@ -0,0 +1,40 @@
+import { db, type IEvent } from './db';
+
+export const getEventReplies = async (eventId: string) => {
+	return await db.events
+		.where('tags')
+		.startsWithIgnoreCase(`e»${eventId}`)
+		.distinct()
+		.and((it) => it.kind === 1)
+		.and((it) => {
+			const lastReply = it.tags
+				.slice()
+				.reverse()
+				.find((tag) => !tag.includes('root') && tag.match(new RegExp(`e»${eventId}.*`, 'g'))?.[0]);
+			return Boolean(lastReply);
+		})
+		.toArray();
+};
+
+export const getEventRoot = async ({ event, eventId }: { event?: IEvent; eventId?: string }) => {
+	if (!event && eventId) event = await db.events.get(eventId);
+	if (!event) return;
+	const eTags = event.tags.filter((it) => it.startsWith('e»'));
+	const rootTag = eTags.find((it) => it.includes('root'));
+	console.log(rootTag, eTags);
+	const id = rootTag ? rootTag.split('»', 3)[1] : eTags?.[0]?.split('»', 3)[1];
+	if (!id) return;
+	return await db.events.get(id);
+};
+
+export const getEventParent = async ({ event, eventId }: { event?: IEvent; eventId?: string }) => {
+	if (!event && eventId) event = await db.events.get(eventId);
+	if (!event) return;
+	const match = event.tags
+		.slice()
+		.reverse()
+		.find((it) => it.startsWith('e»'));
```

Comment:
This ... picks the last e-tag and assumes that's the parent? I'm pretty sure there must be plenty of examples where that is not correct. Marking as "root" or "reply" was introduced to make sorting irrelevant. If one of the tags is marked "reply", it might well be the first or last of several e-tags.

(It might well be that clients still create positional e-tags while marking them root/reply but that might also change. We should probably stick to it but not assume others will forever.)

Reply:
Also trying this branch out with [this event](http://localhost:5173/event/553af9765d12ab18a36d1300d71db34f8ac48275a4a2d952eb890b5f51df78db/) for example, it doesn't give me the replies. [This](http://localhost:5173/event/692faa9aa5f541c41da7b7855c2d988509060492fb8b3640477f2ae3e86f0442/) for example.